### PR TITLE
Add examples with Content-Digest

### PR DESCRIPTION
**Pull request by ioggstream**
## This PR

- Add examples for Content-Digest
- add IANA for Content-Digest
- use "Digest fields" instead of `Digest` and `Content-Digest`
- contentMD5 negotiation replaced by Want-Content-Digest
- fix typos (eg. Location header in a request)

### CODE REVIEW

**Comment by LPardue** [comment]
As a general comment, I'm not a fan of using "Digest fields" instead of Digest and Content-Digest. It introduces a new term of reference and makes things ambiguous. Especially for the text rendering (see #1558 and https://trac.ietf.org/trac/xml2rfc/ticket/647#ticket) a reader is going to have a tough time to understand if

> Digest fields

means multiple instances of the Digest field, or the broader term that is being introduced. For the cost of a few characters, I'd rather use the header names explicitly.

**Comment by ioggstream** [comment]
> , I'd rather use the header names explicitly.

Ok. The point was that the spec name is `digest headers` whereas a more correct name would be `digest fields`. 

Do you see other issues in this PR?

**Comment by LPardue** [comment]
Since the name was decided before adding content-digest, we can consider renaming the document.

There's a few more bits I'll comment on before the end of the day

File: draft-ietf-httpbis-digest-headers.md [hunk 1]
```diff
@@ -117,20 +117,25 @@ HTTP messages are transferred between endpoints, the protocol might choose to
 make use of features of the lower layer in order to provide some integrity
 protection; for instance TCP checksums or TLS records [RFC2818].
 
-This document defines the Digest HTTP integrity mechanism that acts on
-representation data. It operates independent of transport integrity, offering
+This document defines the representation digest integrity mechanism that acts on
+representation data
+and the content digest integrity mechanism that acts instead
+on the conveyed content.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:123]:
```suggestion
This document defines two digest integrity mechanisms for HTTP.
First, representation data integrity, which acts on representation data.
Second, content digest integrity, which acts on conveyed content.
```

File: draft-ietf-httpbis-digest-headers.md [hunk 2]
```diff
@@ -117,20 +117,25 @@ HTTP messages are transferred between endpoints, the protocol might choose to
...
+They operate independent of transport integrity, offering
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:124]:
```suggestion
Both mechanisms operate independent of transport integrity, offering
```

File: draft-ietf-httpbis-digest-headers.md [hunk 3]
```diff
@@ -638,21 +645,26 @@ A user agent supporting both mechanisms:
...
 (one line per key-value pair) with 2 spaced of leading indentation.
 
-`Digest` is media-type agnostic
-and does not provide canonicalization algorithms for specific formats.
-Examples of `Digest` are calculated inclusive of any space.
+Checksum mechanisms defined in this document are media-type agnostic
+and do not provide canonicalization algorithms for specific formats.
+Examples are calculated inclusive of any space.
+While examples can include both fields,
+`Digest` and `Content-Digest` and can be returned independently.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:661]:
```suggestion
`Digest` and `Content-Digest` can be returned independently.
```

File: draft-ietf-httpbis-digest-headers.md [hunk 4]
```diff
@@ -638,21 +645,26 @@ A user agent supporting both mechanisms:
...
 
 ## Server Returns Full Representation Data {#example-full-representation}
 
+In this example, the message content conveys a complete representation data,
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:665]:
```suggestion
In this example, the message content conveys complete representation data,
```

File: draft-ietf-httpbis-digest-headers.md [hunk 5]
```diff
@@ -679,6 +692,7 @@ of a resource.
 The response `Digest` field-value is calculated over the JSON object
 `{"hello": "world"}`, which is not shown because there is no payload
 data.
+`Content-Digest` is computed on an empty content.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:695]:
```suggestion
`Content-Digest` is computed on empty content.
```

File: draft-ietf-httpbis-digest-headers.md [hunk 6]
```diff
@@ -694,14 +708,16 @@ Response:
...
 
 ~~~
 
 ## Server Returns Partial Representation Data
 
 In this example, the client makes a range request and the server
 responds with partial content. The `Digest` field-value represents
-the entire JSON object `{"hello": "world"}`.
+the entire JSON object `{"hello": "world"}`
+while the `Content-Digest` one is computed on the message content `"hello"`.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:720]:
```suggestion
, while the `Content-Digest` field-value is computed on the message content `"hello"`.
```

File: draft-ietf-httpbis-digest-headers.md [hunk 7]
```diff
@@ -892,7 +910,6 @@ Content-Type: application/json
 Accept: application/json
 Accept-Encoding: identity
 Digest: sha-256=bWopGGNiZtbVgHsG+I4knzfEJpmmmQHf7RHDXA3o1hQ=
-Location: /books/123
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:895]: Is this intended? If so, the prose above also needs to be changed to remove reference to Location.

File: draft-ietf-httpbis-digest-headers.md [hunk 8]
```diff
@@ -1034,16 +1051,17 @@ Digest: sha-256=X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=
 
 The following examples demonstrate interactions where a client solicits a
 `Digest` using `Want-Digest`.
+The behavior of `Content-Digest` and `Want-Content-Digest` is analogous.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:1054]:
```suggestion
The behavior of `Content-Digest` and `Want-Content-Digest` is identical.
```

File: draft-ietf-httpbis-digest-headers.md [hunk 9]
```diff
@@ -1153,49 +1171,49 @@ certain identification of the origin of a message [NIST800-32]. Such signatures
...
+Before sending digest fields in a trailer section, the sender
 should consider that intermediaries are explicitly allowed to drop any trailer
 (see Section 6.5.2 of {{SEMANTICS}}).
 
-When `Digest` is used in trailer fields, the receiver gets the digest value after the content
-and may thus be tempted to process the data before validating the digest value.
-It is preferable that data is only be processed after validating the Digest.
+When digest fields are used in a trailer section, the receiver gets there value after the content
+and may thus be tempted to process the data before validating their value.
+It is preferable that data is only be processed after validation.
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:1204]:
```suggestion
When digest fields are used in a trailer section, the field-values are received after the content.
Eager processing of content before the trailer section prevents digest validation, possibly leading to
processing of invalid data.
```

File: draft-ietf-httpbis-digest-headers.md [hunk 10]
```diff
@@ -1296,6 +1314,28 @@ Status:  permanent
 
 Specification document(s):  {{digest}} of this document
 
+## Want-Content-Digest Field Registration
```
Comment (LPardue) [inline draft-ietf-httpbis-digest-headers.md:1317]: I think this is ok for the time being. I'll follow up with an editorial suggestion to the IANA section once we land these changes to main branch.

**Review by LPardue — CHANGES_REQUESTED** [review]
Mainly some nits and editorial suggestions

File: draft-ietf-httpbis-digest-headers.md (hunk 7 above)
Comment (ioggstream) [reply draft-ietf-httpbis-digest-headers.md:895]: @LPardue this was a typo because `Location` is a response header, while this is the request. Is that ok?

**Comment by ioggstream** [comment]
@LPardue all suggestions approved. See comments for the rest. Can we merge?

File: draft-ietf-httpbis-digest-headers.md (hunk 7 above)
Comment (LPardue) [reply draft-ietf-httpbis-digest-headers.md:895]: Yep, all good

**Review by LPardue — APPROVED** [review]

**Pull request merged**